src/main/java/com/influxdb/v3/client/internal/NanosecondConverter.java

```diff
@@ -111,4 +118,49 @@ public static BigInteger convert(final Instant instant, final WritePrecision pre
 
         return FROM_NANOS.get(precision).apply(nanos);
     }
+
+    public static BigInteger getTimestampNano(@Nonnull final Object value, @Nonnull final Field schema) {
+        BigInteger result = null;
+
+        if (value instanceof Long) {
+            if (schema.getFieldType().getType() instanceof ArrowType.Timestamp) {
+                ArrowType.Timestamp type = (ArrowType.Timestamp) schema.getFieldType().getType();
+                TimeUnit timeUnit;
+                switch (type.getUnit()) {
+                    case SECOND:
+                        timeUnit = TimeUnit.SECONDS;
+                        break;
+                    case MILLISECOND:
+                        timeUnit = TimeUnit.MILLISECONDS;
+                        break;
+                    case MICROSECOND:
+                        timeUnit = TimeUnit.MICROSECONDS;
+                        break;
+                    default:
+                    case NANOSECOND:
```

Comment:
`default` should be last
```suggestion
                    case NANOSECOND:
                    default:
```